code/modules/mob/living/simple_animal/abnormality/waw/parasite_tree.dm

```diff
@@ -0,0 +1,382 @@
+
+/mob/living/simple_animal/hostile/abnormality/parasite_tree/proc/endBreach()
+	if(locate(THE_LIARS_BLESSING) in blessed)
+		for(var/datum/status_effect/parasite_tree_blessing/P in blessed)
+			P.facadeFalls()
+		return FALSE
+	if(!minions.len && !locate(THE_TREE_CURSE) in blessed) //no minions? no blessed?
+		cut_overlays()
+		resetQliphoth() //return to non breached sprite while still in containment.
+		return TRUE
+
+/mob/living/simple_animal/hostile/abnormality/parasite_tree/proc/dropLeaf()
+	SIGNAL_HANDLER
+
+	if(origin_cooldown <= world.time && datum_reference.qliphoth_meter > 0) //cool it on the leaf dropping if your already breaching
+		var/list/potentialFollowers = list()
+		origin_cooldown = world.time + (10 SECONDS)
+		for(var/mob/living/carbon/human/L in GLOB.player_list)
+			if(!faction_check_mob(L) && L.stat != DEAD && L.z == z)
+				potentialFollowers += L
+				potentialFollowers[L] = 1
+				if(L.health < (L.maxHealth * 0.6) || L.sanityhealth < (L.maxSanity * 0.6))
+					potentialFollowers[L] += (L.health - 1) + (L.sanityhealth - 1)
+		if(potentialFollowers.len)
+			var/mob/living/carbon/human/chosen_agent = pickweight(potentialFollowers)
+			to_chat(chosen_agent, "<span class='nicegreen'>A large leaf lands nearby.</span>")
+			var/list/possibleleafturf = list()
+			for(var/turf/T in oview(3, chosen_agent))
+				if(!T.density && !locate(/obj/structure/window || /obj/machinery/door) in T.contents)
+					possibleleafturf += T
+			if(possibleleafturf.len > 8) //if your in a area with less than 8 steps of space then theres no room for a leaf
+				new /obj/structure/liars_leaf(pick(possibleleafturf))
+
+	//SAPLING MINION
+/mob/living/simple_animal/hostile/parasite_tree_sapling
+	name = "toxic sapling"
+	desc = "A humanoid tree, it spews thick noxious gas from its agonized face."
+	icon = 'ModularTegustation/Teguicons/32x48.dmi'
+	icon_state = "sapling"
+	icon_living = "sapling"
+	maxHealth = 800
+	health = 800
+	can_patrol = FALSE
+	wander = 0
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1.2, WHITE_DAMAGE = 0.6, BLACK_DAMAGE = 0.8, PALE_DAMAGE = 0.8)
+	ranged = TRUE
+	ranged_cooldown_time = 1 SECONDS
+	obj_damage = 0
+	del_on_death = TRUE
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	deathmessage = "shatters into numerous spongy splinters."
+	deathsound = 'sound/creatures/venus_trap_death.ogg'
+	attacked_sound = 'sound/creatures/venus_trap_hurt.ogg'
+	projectilesound = 'sound/machines/clockcult/steam_whoosh.ogg'
+	var/mob/living/simple_animal/hostile/abnormality/parasite_tree/connected_abno
+
+/mob/living/simple_animal/hostile/parasite_tree_sapling/Initialize()
+	..()
+	icon_living = "sapling[pick(1,2)]"
+	icon_state = icon_living
+	if(locate(/mob/living/simple_animal/hostile/abnormality/parasite_tree) in GLOB.abnormality_mob_list) //this is hopefully a short list.
+		connected_abno = locate(/mob/living/simple_animal/hostile/abnormality/parasite_tree) in GLOB.abnormality_mob_list
+		connected_abno.minions += src
```

Comment:
```suggestion
	connected_abno = locate(/mob/living/simple_animal/hostile/abnormality/parasite_tree) in GLOB.abnormality_mob_list
	if(connected_abno)
		connected_abno.minions += src
```
It is a short list! that's why I made it. connected_abno should null if there isn't a relevant boi, meaning the if-statement covers it, no need to look twice!